sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala

```diff
@@ -277,20 +277,34 @@ class SQLMetricsSuite extends SharedSparkSession with SQLMetricsTestUtils
     withTempView("testDataForJoin") {
       // Assume the execution plan is
       // ... -> SortMergeJoin(nodeId = 1) -> TungstenProject(nodeId = 0)
-      val query = "SELECT * FROM testData2 JOIN testDataForJoin ON testData2.a = testDataForJoin.a"
+      val query1 = "SELECT * FROM testData2 JOIN testDataForJoin ON testData2.a = testDataForJoin.a"
       Seq((0L, 2L, false), (1L, 4L, true)).foreach { case (nodeId1, nodeId2, enableWholeStage) =>
-        val df = spark.sql(query)
+        val df = spark.sql(query1)
         testSparkPlanMetrics(df, 1, Map(
           nodeId1 -> (("SortMergeJoin", Map(
             // It's 4 because we only read 3 rows in the first partition and 1 row in the second one
-            "number of output rows" -> 4L))),
+            "number of output rows" -> 4L,
+            "number of matched rows" -> 4L))),
           nodeId2 -> (("Exchange", Map(
             "records read" -> 4L,
             "local blocks read" -> 2L,
             "remote blocks read" -> 0L,
             "shuffle records written" -> 2L)))),
           enableWholeStage
         )
+
+        val query2 = "SELECT * FROM testData2 JOIN testDataForJoin ON " +
```

Comment:
Why this is inside the `foreach` of line 281?

Reply:
Yes, that was a mistake. Fixed in the latest commit.